src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs

```diff
@@ -2499,9 +2499,6 @@ private bool TryProcessDone(SqlCommand cmd, SqlDataReader reader, ref RunBehavio
             ushort status;
             int count;
 
-            // Can't retry TryProcessDone
-            stateObj._syncOverAsync = true;
```

Comment:
Why this change specifically? Framework TdsParser also has this line

Reply:
> I checked, and TryProcessDone() was not retried after my change. However, I have just become curious what happens if I force it to retry TryProcessDone() by following code:

I meant , are there code paths which could retry Done token processing in certain scenarios for a single SqlDataReader Read operation, instead of multiple Read() operations, which needs to be prevented?

Reply:
@saurabh500 No, I could not find any code path that retries `DONE` token processing in single read operation.

Reply:
@saurabh500 
`TryProcessDone()` runs multiple times until `ReadSni()` actually returns packets, and it is processed by `TryProcessDone()`. Once `DONE` token is processed, codeflow moves forward, and there is no code path to ***RETRY*** the `DONE` token processing, which is already done.